k8s/crds/agentendpoint.mdx

```diff
@@ -302,6 +302,316 @@ When the `spec.upstream.url` has an `https://` scheme, the client certificates w
+
+**Fields:**
+- `name` - Name of the Domain resource
+- `namespace` - Namespace of the Domain resource
+
+**Example:**
+```yaml
+domainRef:
+  name: example-agent-endpoint-ngrok-app
+  namespace: default
+```
+
+### `status.conditions`
+
+Standard Kubernetes conditions that indicate the state of the agent endpoint.
+
+**Type:** `[]Condition`
+
+Each condition includes:
+- `type` - The condition type
+- `status` - `True`, `False`, or `Unknown`
+- `reason` - A programmatic identifier for the condition state
+- `message` - A human-readable description
+- `lastTransitionTime` - When the condition last changed
+- `observedGeneration` - The resource generation this condition applies to
+
+#### Condition Types
+
+##### `Ready`
+
+Indicates whether the agent endpoint is fully operational and ready to receive traffic.
+
+**Status Values:**
+- `True` - Agent endpoint is active and ready
+- `False` - Agent endpoint is not ready (see reason for details)
+
+**Common Reasons:**
+
+| Reason | Status | Meaning |
+|--------|--------|---------|
+| `EndpointActive` | `True` | Agent endpoint is successfully created and active |
+| `Reconciling` | `False` | Endpoint is being reconciled/updated |
+| `Pending` | `False` | Endpoint creation is pending |
+| `Unknown` | `False` | Endpoint state is unknown |
+| `DomainNotReady` | `False` | Associated domain is not ready |
+
+Additionally, the `Ready` condition may reflect reasons from the associated Domain resource when domain issues prevent endpoint creation.
+
+**Example:**
+```yaml
+conditions:
+- type: Ready
+  status: "True"
+  reason: EndpointActive
+  message: "AgentEndpoint is active and ready"
+  lastTransitionTime: "2025-10-29T00:39:26Z"
+  observedGeneration: 1
+```
+
+##### `EndpointCreated`
+
+Indicates whether the endpoint was successfully created in the ngrok API.
+
+**Status Values:**
+- `True` - Endpoint was successfully created
+- `False` - Failed to create endpoint
+
+**Common Reasons:**
+
+| Reason | Status | Meaning |
+|--------|--------|---------|
+| `EndpointCreated` | `True` | Endpoint successfully created |
+| `NgrokAPIError` | `False` | API call to create endpoint failed |
+| `ConfigurationError` | `False` | Endpoint configuration is invalid |
+| `UpstreamError` | `False` | Upstream configuration is invalid |
+
+**Example:**
+```yaml
+conditions:
+- type: EndpointCreated
+  status: "True"
+  reason: EndpointCreated
+  message: "Endpoint successfully created"
+  lastTransitionTime: "2025-10-29T00:39:26Z"
+  observedGeneration: 1
+```
+
+##### `TrafficPolicyApplied`
+
+Indicates whether the traffic policy was successfully applied to the endpoint.
+
+**Status Values:**
+- `True` - Traffic policy is applied
+- `False` - Failed to apply traffic policy
+
+**Common Reasons:**
+
+| Reason | Status | Meaning |
+|--------|--------|---------|
+| `TrafficPolicyApplied` | `True` | Traffic policy successfully applied |
+| `TrafficPolicyError` | `False` | Error applying or validating traffic policy |
+
```

Comment:
This section feels like it's leaving people hanging a bit, since it says "Common Reasons" but doesn't actually list a reason there would be an error applying or validating a traffic policy. Maybe we could list one or two common issues people tend to have?